lib/src/ast.dart

```diff
@@ -9,15 +17,18 @@ abstract class Query {
   /// testing unambiguous.
   @override
   String toString({bool debug = false});
+
+  R cast<R extends Query>() => this as R;
 }
 
 /// Text query to match [text].
 ///
 /// [isExactMatch] is set when the [text] was inside quotes.
-class TextQuery implements Query {
+class TextQuery extends Query {
   final String text;
   final bool isExactMatch;
-  TextQuery(this.text, {this.isExactMatch = false});
+  const TextQuery(this.text, super.startIndex, super.endIndex,
```

Comment:
Fixed position arguments make sense when the list is short and self-evident, but these extra parameters push it beyond the self-evident range. I think it would be better to make them named parameters + `required`, wdyt?

Reply:
Works for me